File: server/src/main/java/com/example/schoolsite/services/impl/SheduleServiceImpl.java
package com.example.schoolsite.services.impl;


import com.example.schoolsite.dto.SheduleDTO;
import com.example.schoolsite.entity.*;
import com.example.schoolsite.exception.ResourceNotFoundException;
import com.example.schoolsite.map.Mapper;
import com.example.schoolsite.services.SheduleService;
import com.example.schoolsite.workWithDatabase.repo.*;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class SheduleServiceImpl implements SheduleService {

    private final AttendanceRepository attendanceRepository;

    private final AcademicPerfomanceRepository academicPerfomanceRepository;

    private final PupilRepository pupilRepository;

    private final SheduleRepository sheduleRepository;

    private final SubjectRepository subjectRepository;

    private final ClassroomRepository classroomRepository;

    private final TeacherRepository teacherRepository;

    private final CalendarRepository calendarRepository;

    @Override
    public List<SheduleDTO> getScheduleByIdAndDate(@PathVariable(value = "userId") Long userId, @PathVariable(value = "date") String date) {
        return getScheduleDTOByIdAndDate(userId, date);
    }

    @Override
    public List<SheduleDTO> getScheduleDTOByIdAndDate(Long userId, String date) {
        Pupil pupil = pupilRepository.findByUserId(userId);
        List<Shedule> sheduleList = sheduleRepository.findAllByClassroomIDAndDate(pupil.getClassroomId().getId(), Date.valueOf(date));
        List<SheduleDTO> sheduleDTOList = new ArrayList<>();

        Classroom classroom = classroomRepository.getById(pupil.getClassroomId().getId());

        for (Shedule shedule : sheduleList) {
            Subject subject = subjectRepository.getById(shedule.getSubjectID().getId());
            Teacher teacher = teacherRepository.getById(shedule.getTeacherID().getId());
            Calendar calendar = calendarRepository.getById(shedule.getCalendarId().getId());
            SheduleDTO sheduleDTO = Mapper.mapSheduleToSheduleDTO(shedule, calendar, subject, teacher, classroom);
            sheduleDTOList.add(sheduleDTO);
        }
        return sheduleDTOList;
    }

}

File: server/src/main/java/com/example/schoolsite/services/impl/SheduleServiceImpl.kt
package com.example.schoolsite.services.impl

import com.example.schoolsite.dto.SheduleDTO

@Service
@RequiredArgsConstructor
class SheduleServiceImpl : SheduleService {
    private val attendanceRepository: AttendanceRepository? = null
    private val academicPerfomanceRepository: AcademicPerfomanceRepository? = null
    private val pupilRepository: PupilRepository? = null
    private val sheduleRepository: SheduleRepository? = null
    private val subjectRepository: SubjectRepository? = null
    private val classroomRepository: ClassroomRepository? = null
    private val teacherRepository: TeacherRepository? = null
    private val calendarRepository: CalendarRepository? = null
    @Override
    fun getScheduleByIdAndDate(@PathVariable(value = "userId") userId: Long?, @PathVariable(value = "date") date: String?): List<SheduleDTO> {
        return getScheduleDTOByIdAndDate(userId, date)
    }

    @Override
    fun getScheduleDTOByIdAndDate(userId: Long?, date: String?): List<SheduleDTO> {
        val pupil: Pupil = pupilRepository.findByUserId(userId)
        val sheduleList: List<Shedule> = sheduleRepository.findAllByClassroomIDAndDate(pupil.getClassroomId().getId(), Date.valueOf(date))
        val sheduleDTOList: List<SheduleDTO> = ArrayList()
        val classroom: Classroom = classroomRepository.getById(pupil.getClassroomId().getId())
        for (shedule in sheduleList) {
            val subject: Subject = subjectRepository.getById(shedule.getSubjectID().getId())
            val teacher: Teacher = teacherRepository.getById(shedule.getTeacherID().getId())
            val calendar: Calendar = calendarRepository.getById(shedule.getCalendarId().getId())
            val sheduleDTO: SheduleDTO = Mapper.mapSheduleToSheduleDTO(shedule, calendar, subject, teacher, classroom)
            sheduleDTOList.add(sheduleDTO)
        }
        return sheduleDTOList
    }
}